Source/Events.Processing/Filters/ValidateFilterByComparingStreams.cs

```diff
@@ -21,105 +19,85 @@ namespace Dolittle.Runtime.Events.Processing.Filters
     public class ValidateFilterByComparingStreams : IValidateFilterByComparingStreams
     {
         readonly IEventFetchers _eventFetchers;
-        readonly IStreamProcessorStateRepository _streamProcessorStates;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ValidateFilterByComparingStreams"/> class.
         /// </summary>
         /// <param name="eventFetchers">The <see cref="IEventFetchers" />.</param>
-        /// <param name="streamProcessorStates">The <see cref="IStreamProcessorStateRepository" />.</param>
         public ValidateFilterByComparingStreams(
-            IEventFetchers eventFetchers,
-            IStreamProcessorStateRepository streamProcessorStates)
+            IEventFetchers eventFetchers)
         {
             _eventFetchers = eventFetchers;
-            _streamProcessorStates = streamProcessorStates;
         }
 
         /// <inheritdoc/>
-        public async Task<FilterValidationResult> Validate<TFilterDefinition>(IFilterDefinition persistedDefinition, IFilterProcessor<TFilterDefinition> filter, CancellationToken cancellationToken)
+        public async Task<FilterValidationResult> Validate<TFilterDefinition>(IFilterDefinition persistedDefinition, IFilterProcessor<TFilterDefinition> filter, StreamPosition lastUnprocessedEvent, CancellationToken cancellationToken)
             where TFilterDefinition : IFilterDefinition
         {
-            if (persistedDefinition == default) return new FilterValidationResult();
-
-            if (persistedDefinition.Partitioned != filter.Definition.Partitioned)
-            {
-                return new FilterValidationResult($"The new stream generated from the filter will not match the old stream. {(persistedDefinition.Partitioned ? "The previous filter is partitioned while the new filter is not" : "The previous filter is not partitioned while the new filter is")}");
-            }
-
-            if (persistedDefinition.Public != filter.Definition.Public)
-            {
-                return new FilterValidationResult($"The new stream generated from the filter will not match the old stream. {(persistedDefinition.Public ? "The previous filter is public while the new filter is not" : "The previous filter is not public while the new filter is")}");
-            }
-
-            var tryGetState = await _streamProcessorStates.TryGetFor(
-                new StreamProcessorId(filter.Scope, filter.Definition.TargetStream.Value, filter.Definition.SourceStream),
-                cancellationToken)
-                .ConfigureAwait(false);
-            if (!tryGetState.Success)
+            try
```

Comment:
We can come back to this method later since it in my opinion should be made more readable by putting more things in methods :)